contracts/src/utils/nonces.rs

```diff
@@ -0,0 +1,139 @@
+//! Nonces Contract
+//!
+//! Contract module which provides functionalities for tracking nonces for
+//! addresses.
+//!
+//! Note: Nonce will only increment.
+
+use alloy_primitives::{Address, U256};
+use alloy_sol_types::sol;
+use stylus_proc::{external, sol_storage, SolidityError};
+
+sol! {
+    /// The nonce used for an `account` is not the expected current nonce.
+    #[derive(Debug)]
+    #[allow(missing_docs)]
+    error InvalidAccountNonce(address account, uint256 currentNonce);
+}
+
+/// A Nonces error.
+#[derive(SolidityError, Debug)]
+pub enum Error {
+    /// The nonce used for an `account` is not the expected current nonce.
+    InvalidAccountNonce(InvalidAccountNonce),
+}
+
+sol_storage! {
+    /// State of a Nonces Contract.
+    #[cfg_attr(all(test, feature = "std"), derive(motsu::DefaultStorageLayout))]
+    pub struct Nonces {
+        /// Mapping from address to its nonce.
+        mapping(address => uint256) _nonces;
+    }
+}
+
+#[external]
+impl Nonces {
+    /// Returns the unuse nonce for the given `account`.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    /// - `owner` - The address for which to return the nonce.
+    fn nonce(&self, owner: Address) -> U256 {
+        self._nonces.get(owner)
+    }
+
+    /// Consumes a nonce for the given `account`.
+    ///
+    /// # Arguments
+    ///
+    /// * `&mut self` - Write access to the contract's state.
+    /// - `owner` - The address for which to consume the nonce.
+    ///
+    /// * Returns the current nonce for the given `account` and increments the
+    ///   nonce.
+    fn use_nonce(&mut self, owner: Address) -> Result<U256, Error> {
+        let nonce = self._nonces.get(owner);
+        self._nonces.setter(owner).set(nonce + U256::from(1u32));
```

Comment:
Let's extract `U256::from(1)` as a const with the `uint!` macro, and then use `nonce.checked_add(1).unwrap_unchecked()` with the appropriate safety comment, which you can reuse from the Solidity reference.